### DISCUSSION TITLE
How to use with different node versions?

### DISCUSSION THREAD

**Discussion by ErlanBazarov** [vercel/turborepo#1525] [Help] [answered]
I need to use one node version for a workspace, and a different node version for another workspace. How to do this appropriately? `npm install -w=workspace` is throwing error because of package.json of another workspace which needs different node version

**Comment by rafaeltab** [comment, marked as answer]
I don't believe this is possible. Node is installed globally across your entire system. If you do want this, you'll have to switch node versions before running the different workspaces. This also means you can't run them simultaneously. NOTE: This isn't a limitation set by turborepo, and it would be impossible for turborepo to add this functionality.

**Reply by mryechkin** [reply]
Agreed - this sounds like all sorts of problems waiting to happen, Turborepo or not.

I'm curious what the use case here is. I can't think of a scenario where I'd want to have this in a monorepo ever. If a project can only run on a specific version of Node, different from all others, then it doesn't belong in a monorepo IMO

**Comment by keinsell** [comment]
I'm not sure but maybe `volta` can help you a bit. I have no clue this will work in monorepository but you can try.
